x11-packages/rnote/build.sh

```diff
@@ -8,25 +8,38 @@ TERMUX_PKG_SHA256=b133d4331963d3c09d3a7477f60fc4c5072471dcbf459379a593ca1724164a
 TERMUX_PKG_AUTO_UPDATE=true
 TERMUX_PKG_DEPENDS="gdk-pixbuf, gettext, glib, graphene, gtk4, hicolor-icon-theme, libadwaita, libcairo, pipewire, pango, poppler"
 TERMUX_PKG_BUILD_DEPENDS="libiconv"
+TERMUX_PKG_PYTHON_BUILD_DEPS="toml2json"
 
 __fetch_gettext_rs() {
-    # This version is relative to gettext-sys provided by the gettext-rs crate
-	local _version="0.21.4"
-	local _url="https://github.com/gettext-rs/gettext-rs/archive/refs/tags/gettext-sys-$_version.tar.gz"
-	local _path="$TERMUX_PKG_CACHEDIR/gettext-v$_version.tar.gz"
-	termux_download "$_url" "$_path" SKIP_CHECKSUM
-	tar xf "$_path" -C "$TERMUX_PKG_SRCDIR"
-	mv "gettext-rs-gettext-sys-$_version" "gettext-source"
+	# This version is relative to gettext-sys provided by the gettext-rs crate
+	local crate_version
+	crate_version="$(curl -s https://crates.io/api/v1/crates/gettext-sys | jq --raw-output '.crate.max_stable_version')"
+	grep -qP '\d+\.\d+\.\d+' <<< "$crate_version" || {
+		termux_error_exit "Unable to fetch latest version for \`gettext-sys\` crate"
+	}
+	local -a crate=(
+		"https://github.com/gettext-rs/gettext-rs/archive/refs/tags/gettext-sys-$crate_version.tar.gz" # Upstream URL
+		"$TERMUX_PKG_CACHEDIR/gettext-v$crate_version.tar.gz"                                          # Local save path
+		'211773408ab61880b94a0ea680785fc21fad307cd42594d547cf5a056627fcda'                             # SHA256 checksum
+	)
+	# Fetch latest and greatest gettext from upstream and place it inside gettext-source/gettext-sys
+	local -a upstream=(
+		"$(. "$TERMUX_SCRIPTDIR/packages/gettext/build.sh"; echo ${TERMUX_PKG_SRCURL})" # Upstream URL (from gettext's build script)
+		"$TERMUX_PKG_SRCDIR/gettext-source/gettext-sys"                                 # Local save path
```

Comment:
Derp, switching to `termux_download` means we're using `curl` in the background now, meaning we need to specify the filename.
Never let the build get far enough to notice when testing locally because I was just worried about it downloading fine in the first place, didn't think about the filename.